dev_guide/secrets.adoc

```diff
@@ -119,9 +119,20 @@ Repositories for Builds] for more information.
 [[secrets-restrictions]]
 == Restrictions
 
-Secret volume sources are validated to ensure that the specified object
-reference points to a `*Secret*` object. Therefore, a secret needs to be created
-before the pods that depend on it.
+To use a secret, a pod needs to reference the secret. A secret can be used with
+a pod in two ways: either as files in a volume mounted on one or more of its
+containers, or used by kubelet when pulling images for the pod.
+
+Volume type secrets write data into the container as a file using the volume
+mechanism. *imagePullSecrets* use service accounts for the automatic injection of
+the secret into all pods in a namespaces.
+
+When a template contains a secret definition, the only way for the template to
+use the provided secret is to ensure that the secret volume sources are
+validated to ensure that the specified object reference actually points to an
+object of type `*Secret*`. Therefore, a secret needs to be created before any pods
+that depend on it. Th most effective way to ensure this is to have it
```

Comment:
- s/Th/The/
- Too much "ensure" for me...  If A ensures that EVENT happens, we can try "For EVENT, A must...".